File: code/mathematical_algorithms/src/gcd_and_lcm/gcd_and_lcm.scala
object gcdandlcm extends App{
	def gcd(a:Int, b:Int):Int = {
		while(b > 0){
			val r:Int = a % b
			a = b
			b = r
		}
		a
	}

	def lcm(a:Int, b:Int):Int = {
		var num1:Int = 0
		var num2:Int = 0
		var lcm:Int = 0
		if(a > b){
			num1 = a
			num2 = b
		} else {
			num1 = b
			num2 = a
		}
		for(x <- 1 to num2){
			if((num1 * i) % num2 == 0){
				return num1 * i
			}
		}
		return 0
	}
}

File: code/mathematical_algorithms/src/gcd_and_lcm/gcd_and_lcm.cs
using System;

namespace gcd_and_lcm
{
    class gcd_lcm
    {
        int a, b;
        public gcd_lcm(int number1,int number2)
        {
            a = number1;
            b = number2;
        }
        public int gcd()
        {
            int temp1 = a, temp2 = b;
            while (temp1 != temp2)
            {
                if (temp1 > temp2)
                {
                    temp1 -= temp2;
                }
                else
                {
                    temp2 -= temp1;
                }
            }
            return temp2;
        }
        public int lcm()
        {
            return a * b / gcd();
        }
    }
    class Program
    {
        static void Main(string[] args)
        {
            int a = 20, b = 120;
            gcd_lcm obj = new gcd_lcm(a, b);

            Console.WriteLine("GCD of {0} and {1} is {2}", a, b, obj.gcd());
            Console.WriteLine("LCM of {0} and {1} is {2}", a, b, obj.lcm());

            Console.ReadKey();
        }
    }
}
